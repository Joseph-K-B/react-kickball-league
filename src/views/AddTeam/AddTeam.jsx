import { useState } from "react"
import { useHistory } from "react-router-dom"

function AddTeam() {
    const [name, setName] = useState('');
    const [city, setCity] = useState('');
    const [state, setState] = useState('');
    const histor = useHistory();

    const handleSubmit = async (e) => {
        e.preventDefault();

        // const res = await createTeam({name, sity, state})
        // history.pushState(`/teams/${res[0].id}`)
    }

    return(
        <>
        <fieldset>
            <legend>Add Team</legend>
            <form>
                <label htmlFor='name'>Name:</label>
                <input id='name' type="text" value={name} onChange={({target}) => setName(target.value)}/>

                <label htmlFor='city'>City:</label>
                <input id='city' type="text" value={city} onChange={({target}) => setCity(target.value)}/>

                <label htmlFor='state'>State:</label>
                <input id='state' type="text" value={state} onChange={({target}) => setState(target.value)}/>

                <button aria-label='add team' type="submit" onSubmit={handleSubmit}>Add</button>
            </form>
        </fieldset>
        </>
    )
}

export default AddTeam;