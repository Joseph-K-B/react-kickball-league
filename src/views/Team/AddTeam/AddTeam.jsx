import { useState } from "react"
import { useHistory } from "react-router-dom"
import TeamForm from "../../../components/TeamForm";
import { createTeam } from "../../../services/teams";

function AddTeam() {
    const [name, setName] = useState('');
    const [city, setCity] = useState('');
    const [state, setState] = useState('');
    const history = useHistory();

    const handleAddSubmit = async (e) => {
        e.preventDefault();
        const res = await createTeam({name, city, state})
        history.push(`/teams/${res[0].id}`)
    }

    return(
        <>
        <fieldset>
            <legend>Add Team</legend>
            <TeamForm 
                name={name}
                city={city}
                state={state}
                handleSubmit={handleAddSubmit}
                setName={setName}
                setCity={setCity}
                setState={setState}
            />
        </fieldset>
        </>
    )
}

export default AddTeam;