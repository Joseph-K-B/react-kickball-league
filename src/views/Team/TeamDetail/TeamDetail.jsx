import { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";
import { getTeamById } from "../../../services/teams";

function TeamDetail() {
    const { teamId } = useParams()
    const[team, setTeam] = useState(null)

    useEffect(() => {
        getTeamById(teamId).then((res) => setTeam(res))
    }, [teamId])

    if(!team) return <h2>Loading team...</h2>

    return(
        <>
        <h1>{team.name}</h1>
        <h3>{team.city}, {team.state}</h3>
        <p>Wins: {team.wins}</p>
        <ul>
        {team.players.map((player) => (
            <li key={player.id}>
                <Link className='link' to={`/players/${player.id}`}>
                    <p>{player.position} - {player.name}</p>
                </Link>
            </li>
        ))}
        </ul>
        <Link className='link' to='/teams'><p>Team Directory</p></Link>
        </>
    )
}

export default TeamDetail