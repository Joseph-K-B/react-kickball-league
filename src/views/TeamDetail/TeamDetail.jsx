import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { getPlayers } from "../../services/players";
import { getTeamById } from "../../services/teams";

function TeamDetail() {
    const { id } = useParams()
    const[team, setTeam] = useState(null)

    useEffect(() => {
        getTeamById(id).then((res) => setTeam(res))
    }, [id])

    if(!team) return <h2>Loading team...</h2>

    return(
        <>
        <h1>{team.name}</h1>
        <h3>{team.city}, {team.state}</h3>
        <ul>
        {team.players.map((player) => (
            <li key={player.id}>{player.position} - {player.name}</li>
        ))}
        </ul>

        </>
    )
}

export default TeamDetail