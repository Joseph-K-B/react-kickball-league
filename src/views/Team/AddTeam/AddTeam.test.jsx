import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import {Router, Route } from 'react-router-dom';
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event';
import AddTeam from './AddTeam';
import TeamDetail from '../TeamDetail/TeamDetail';

const mockTeam = {
        id: 20,
        created_at: '2021-12-08T16:09:33.408898+00:00',
        name: 'name',
        city: 'city',
        state: 'state',
        players: []
}

const server = setupServer(
    rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res(ctx.json(mockTeam));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockTeam]));
    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders form to add team and redirects to team details', async () => {
    const history = createMemoryHistory();
    history.push('/teams/new');

    const {container} = render(
        <Router history={history}>
            <Route exact path='/teams/new'>
                <AddTeam />
            </Route>
            <Route exact path='/teams/:id' component={TeamDetail} />
        </Router>
    );

    screen.getByText('Add Team');

    const nameField = screen.getByLabelText(/Name:/);
    const cityField = screen.getByLabelText(/City:/);
    const stateField = screen.getByLabelText(/State:/);
    const submitBtn = screen.getByRole('button', {name: 'edit team' })

    userEvent.type(nameField, 'Another Team');
    userEvent.type(cityField, 'Another Town');
    userEvent.type(stateField, 'US');
    userEvent.click(submitBtn);

    expect(container).toMatchSnapshot();
    await screen.findByText('name');
});