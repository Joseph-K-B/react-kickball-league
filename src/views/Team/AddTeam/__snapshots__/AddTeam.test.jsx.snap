// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form to add team and redirects to team details 1`] = `
<div>
  <fieldset>
    <legend>
      Add Team
    </legend>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        type="text"
        value="Another Team"
      />
      <label
        for="city"
      >
        City:
      </label>
      <input
        id="city"
        type="text"
        value="Another Town"
      />
      <label
        for="state"
      >
        State:
      </label>
      <input
        id="state"
        type="text"
        value="US"
      />
      <button
        aria-label="edit team"
        type="submit"
      >
        Submit
      </button>
    </form>
  </fieldset>
  <h2>
    Loading team...
  </h2>
</div>
`;
